src/lib/components/ui/ScrollView/ScrollView.jsx

```diff
@@ -75,6 +101,47 @@ export const ScrollView = (props) => {
     debounce,
   );
 
+  const autoScrollChildrenKeys = autoScroll !== 'off' && children && React.Children
+    .map(children, (child) => child.key)
+    .reduce((reducedKeys, childKey) => reducedKeys + childKey, '');
+  const autoScrollChildrenLength = autoScroll !== 'off' && children && children.length;
+
+  /**
+   * If autoScroll is enabled, it automatically scrolls viewport element to the end of the
+   * the viewport when content is changed. It is performed only when viewport element is
+   * scrolled to the end of the viewport or when viewport element is in any position but
+   * autoScroll triggered by previous change is still in progress.
+   */
+  useLayoutEffect(() => {
+    if (autoScroll === 'off') {
```

Comment:
No. Effects must be executed always at the same time and order, otherwise it throws error. ESlint also checks if effect is not in condition. All conditions whether to execute hook or not are passed as deps array into effect - if teey changes, effect is called.